### PR TITLE
V3.0.0 - testnet version

### PR DESCRIPTION
**Pull request by arhamchordia**
## 1. Overview

This PR upgrades the Quasar chain from Cosmos SDK version 0.47 to version 0.50.x. This includes updating dependencies, modifying import paths, and adjusting the codebase to align with the changes and improvements introduced in the newer SDK version.

## 2. Implementation details
Ref for the cosmos-sdk upgrade - https://github.com/cosmos/cosmos-sdk/blob/release/v0.50.x/UPGRADING.md](https://github.com/cosmos/cosmos-sdk/blob/release/v0.50.x/UPGRADING.md

## 3. How to test/use

TODO. 

## 4. Checklist

- [ ] Does the Readme need to be updated? YES

## 5. Limitations (optional)

The upgrade may introduce breaking changes requiring dependent services or module modifications.
Extensive testing is needed to ensure compatibility with existing data and user workflows.

### CODE REVIEW

File: app/keepers/keepers.go [hunk 1]
```diff
@@ -194,35 +198,40 @@ func (appKeepers *AppKeepers) InitNormalKeepers(
...
 	appKeepers.BankKeeper = bankKeeper
 
 	stakingKeeper := stakingkeeper.NewKeeper(
 		appCodec,
-		appKeepers.keys[stakingtypes.StoreKey],
+		runtime.NewKVStoreService(appKeepers.keys[stakingtypes.StoreKey]),
 		appKeepers.AccountKeeper,
 		appKeepers.BankKeeper,
 		authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+		address.NewBech32Codec(sdk.GetConfig().GetBech32AccountAddrPrefix()),
```
Comment (arhamchordia) [inline app/keepers/keepers.go:227]: Concern here as it gives cosmos as prefix instead of quasar

File: app/keepers/keepers.go [hunk 2]
```diff
@@ -153,24 +156,25 @@ func (appKeepers *AppKeepers) InitSpecialKeepers(
...
 	appKeepers.CrisisKeeper = crisiskeeper.NewKeeper(
 		appCodec,
-		appKeepers.keys[crisistypes.StoreKey],
-		//app.GetSubspace(crisistypes.ModuleName),
+		runtime.NewKVStoreService(appKeepers.keys[crisistypes.StoreKey]),
 		invCheckPeriod,
 		appKeepers.BankKeeper,
 		authtypes.FeeCollectorName,
 		authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+		address.NewBech32Codec(sdk.GetConfig().GetBech32AccountAddrPrefix()),
```
Comment (arhamchordia) [inline app/keepers/keepers.go:172]: Gets codec as cosmos

File: app/keepers/keepers.go (hunk 1 above)
Comment (ajansari95) [reply app/keepers/keepers.go:227]: setAddressPrefixes in config in  main.go before svrcmd.Execute

File: proto/quasar/tokenfactory/v1beta1/query.proto [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+option go_package = "github.com/quasar-finance/quasar/x/tokenfactory/types";
+
+// Query defines the gRPC querier service.
+service Query {
+  // Params defines a gRPC query method that returns the tokenfactory module's
+  // parameters.
+  rpc Params(QueryParamsRequest) returns (QueryParamsResponse) {
+    option (google.api.http) = {
+      get: "/quasar/tokenfactory/v1beta1/params"
+      additional_bindings {
```
Comment (akure) [inline proto/quasar/tokenfactory/v1beta1/query.proto:19]: Why is this additional binding required with the previous values proto paths? Is there any constrained do we have.

File: proto/quasar/tokenfactory/v1beta1/tx.proto [hunk 4]
```diff
@@ -44,36 +47,46 @@ message MsgCreateDenomResponse {
...
-  [ (gogoproto.moretags) = "yaml:\"burn_from_address\"" ];
+      [ (gogoproto.moretags) = "yaml:\"burn_from_address\"" ];
 }
 
 message MsgBurnResponse {}
 
 // MsgChangeAdmin is the sdk.Msg type for allowing an admin account to reassign
 // adminship of a denom to a new account
 message MsgChangeAdmin {
+  option (amino.name) = "quasar/tokenfactory/change-admin";
```
Comment (akure) [inline proto/quasar/tokenfactory/v1beta1/tx.proto:88]: Why do we need the amino.name ?

File: proto/quasar/tokenfactory/v1beta1/query.proto (hunk 3 above)
Comment (arhamchordia) [reply proto/quasar/tokenfactory/v1beta1/query.proto:19]: @ajansari95

Comment (ajansari95) [reply proto/quasar/tokenfactory/v1beta1/query.proto:19]: so we are backward compatible with old routes, if somebody has workflow it doesn't break. We can get rid of it later.Or if we dont care about backward compatibility we can remove.

**Review by ajansari95 — APPROVED** [review]
LGTM

**Review by akure — APPROVED** [review]

**Comment by akure** [comment]
Sounds good to me.

**Pull request merged**